crates/ruff_linter/src/rules/flake8_use_pathlib/helpers.rs

```diff
@@ -0,0 +1,72 @@
+use crate::checkers::ast::Checker;
+use crate::importer::ImportRequest;
+use crate::{Applicability, Edit, Fix, Violation};
+use ruff_python_ast::{Expr, ExprCall};
+use ruff_text_size::Ranged;
+
+pub(crate) fn is_path_call(checker: &Checker, expr: &Expr) -> bool {
+    expr.as_call_expr().is_some_and(|expr_call| {
+        checker
+            .semantic()
+            .resolve_qualified_name(&expr_call.func)
+            .is_some_and(|name| matches!(name.segments(), ["pathlib", "Path"]))
+    })
+}
+
+pub(crate) fn check_os_path_get_calls(
+    checker: &Checker,
+    call: &ExprCall,
+    fn_name: &str,
+    attr: &str,
+    fix_enabled: bool,
+    violation: impl Violation,
+) {
+    if checker
+        .semantic()
+        .resolve_qualified_name(&call.func)
+        .is_some_and(|qualified_name| qualified_name.segments() != ["os", "path", fn_name])
```

Comment:
Well this is a bit embarrassing, but I just realized that my suggestion _did_ break your code, and CI was correct. This should be `is_none_or` to preserve the original behavior.

```suggestion
        .is_none_or(|qualified_name| qualified_name.segments() != ["os", "path", fn_name])
```